sdcm/perfomance_tuning_test.py

```diff
@@ -0,0 +1,151 @@
+        raise ValueError(f"Unrecognized number of CPUs: {self.number_of_cpu_cores}")
+
+    def get_expected_options_file_contents(self) -> dict:
+        # Scylla version regex: '(\d+\.\d+)'
+
+        # string = dedent("""cpu_mask: '0x0000000f'
+        #                    irq_core_auto_detection_ratio: 16
+        #                    irq_cpu_mask: '0x0000000f'
+        #                    nic:
+        #                    - <NIC iface name>
+        #                    tune:
+        #                    - net""")
+        base_result_dict = {"cpu_mask": self.get_expected_cpu_mask(),
+                            "tune": ["net"]}
+        if (self.node.is_enterprise and self.comparable_scylla_version >= "2022.2.7)")\
+                or self.comparable_scylla_version >= "5.2":
+            attach_values = {"irq_core_auto_detection_ratio": 16,
+                             "irq_cpu_mask": self.get_expected_irq_cpu_mask(),
+                             "nic": [self.nic_name],
+                             }
+        elif (self.node.is_enterprise and self.comparable_scylla_version >= "2022.1")\
+                or self.comparable_scylla_version >= "5.0":
+            attach_values = {"mode": "mq",
+                             "nic": [self.nic_name],
+                             }
+        elif (self.node.is_enterprise and self.comparable_scylla_version >= "2021.1")\
+                or self.comparable_scylla_version >= "4.6":
+            attach_values = {"mode": "mq",
+                             "nic": self.nic_name,
+                             }
+        else:
+            raise ValueError(f"Unfamiliar scylla version: {self.comparable_scylla_version}")
+        base_result_dict.update(attach_values)
+        return base_result_dict
+
+
+class PerftuneExecutor:
+    def __init__(self, node):
+        self.node = node
+
+    def get_cpu_mask(self) -> str:
+        nic_name = self.node.get_nic_devices()[0]
+        result = self.node.remoter.run(f"{PERFTUNE_LOCATION} --tune net --nic {nic_name} --get-cpu-mask")
+        return result.stdout.strip()
+
+    def get_irq_cpu_mask(self) -> str:
+        nic_name = self.node.get_nic_devices()[0]
+        result = self.node.remoter.run(f"{PERFTUNE_LOCATION} --tune net --nic {nic_name} --get-irq-cpu-mask")
+        return result.stdout.strip()
+
+    def get_options_file_contents(self) -> dict:  # , file_path=None
+        nic_name = self.node.get_nic_devices()[0]
+        result = self.node.remoter.run(f"{PERFTUNE_LOCATION} --tune net --nic {nic_name} –mode mq --dump-options-file")
+        result_as_yaml = yaml.safe_load(result.stdout)
+        return result_as_yaml
+
+
+class PerftuneOutputChecker:  # pylint: disable=too-few-public-methods
+    def __init__(self, node):
+        self.node = node
+        self.executor = PerftuneExecutor(node)
+        self.expected_result = ExpectedResult(node)
+
+    def compare_perftune_results(self) -> None:
```

Comment:
better to put this whole implemetion under exception handling to make sure we won't stop any test cause of those checks

Reply:
I see your point. Done!